lstchain/scripts/lstchain_create_run_summary.py

```diff
@@ -0,0 +1,216 @@
+"""
+Create a run summary for a given date containing the number of subruns,
+the start time of the run, type pf the run: DATA, DRS4, CALI, and
+the reference timestamp and counter of the run.
+"""
+
+import argparse
+import logging
+from pathlib import Path
+
+import numpy as np
+import protozfits
+from ctapipe.containers import EventType
+from ctapipe.io import EventSource
+from ctapipe_io_lst.event_time import CENTRAL_MODULE
+from lstchain.paths import parse_r0_filename
+
+# FIXME: take it from somewhere else
+tel_id = 1
+
+
+log = logging.getLogger(__name__)
+
+parser = argparse.ArgumentParser(description="Create run summary file")
+
+parser.add_argument(
+    "-d",
+    "--date",
+    type=str,
+    help="Date for the creation of the run summary in format YYYYMMDD",
+    required=True,
+)
+# TODO: default today()
+
+parser.add_argument(
+    "--r0-path",
+    type=Path,
+    dest="R0_PATH",
+    help="Path to the R0 files. Default is /fefs/aswg/data/real/R0",
+    default=Path("/fefs/aswg/data/real/R0"),
+)
+
+parser.add_argument(
+    "-o",
+    "--output-dir",
+    type=Path,
+    dest="output_dir",
+    help="Directory in which Run Summary file is written",
+    default="./",
+)
+
+
+args = parser.parse_args()
+
+
+def get_list_of_files(date):
+    """Get the list of R0 files from a given date."""
+    path_r0 = args.R0_PATH / date
+    # FIXME: use regular expressions from lstchain.paths.R0_RE
+    list_of_files = path_r0.glob("LST*.fits.fz")
+    return list_of_files
+
+
+def get_list_of_runs(list_of_files):
+    """Get the sorted list of run objects from R0 filenames."""
+    return sorted(parse_r0_filename(file) for file in list_of_files)
+
+
+def get_run_numbers(list_of_runs):
+    """Get the list of run numbers of a given date."""
+    return np.unique([run_info.run for run_info in list_of_runs])
+
+
+def get_number_of_subruns(run_number, list_of_run_objects):
+    """Obtain the number of sequential files (subruns) of a given run."""
+    filtered_run = filter(lambda x: x.run == run_number, list_of_run_objects)
+    last_subrun = max(list(filtered_run), key=lambda x: x.subrun)
+    return last_subrun.subrun
+
+
+def start_of_run_datetime(run_number):
+    """
+    Get datetime of the start of the run in ISOT format and UTC scale.
+    """
+    # TODO: Use pathlib.stat_results instead to get creation file timestamp instead
+    filename = args.R0_PATH / args.date / f"LST-1.1.Run{run_number:05d}.0000.fits.fz"
+    file = protozfits.File(str(filename))
+    return file.CameraConfig.header["DATE"]
```

Comment:
Is this the same as `file.CameraConfig[1].date`? "DATE" in the header should be the file creation date as UTC ISO timestamp. `file.CameraConfig[1].date` should be the start of run timestamp as unix time.

Reply:
Ok, at least for the one run I checked, seems to be the same:

```
In [5]: f.CameraConfig[0].date
Out[5]: 1582059741

In [6]: f.CameraConfig.header['DATE']
Out[6]: '2020-02-18T21:02:21'

In [7]: from astropy.time import Time

In [8]: Time(f.CameraConfig[0].date, format='unix').isot
Out[8]: '2020-02-18T21:02:21.000'
```

Reply:
Since I'm using the EventSource to get the reference timestamps, I decided to get this `run_date` from there as well `lst.svc.date` instead of using protozfits